presto-verifier/src/main/java/com/facebook/presto/verifier/rewrite/FunctionCallRewriter.java

```diff
@@ -13,51 +13,140 @@
+import com.facebook.presto.metadata.FunctionAndTypeManager;
+import com.facebook.presto.spi.function.FunctionKind;
+import com.facebook.presto.spi.function.Signature;
+import com.facebook.presto.spi.function.SqlFunction;
+import com.facebook.presto.sql.ExpressionFormatter;
+import com.facebook.presto.sql.parser.ParsingException;
+import com.facebook.presto.sql.parser.SqlParser;
+import com.facebook.presto.sql.tree.ArrayConstructor;
+import com.facebook.presto.sql.tree.CurrentTime;
 import com.facebook.presto.sql.tree.Expression;
 import com.facebook.presto.sql.tree.ExpressionRewriter;
 import com.facebook.presto.sql.tree.ExpressionTreeRewriter;
 import com.facebook.presto.sql.tree.FunctionCall;
+import com.facebook.presto.sql.tree.Identifier;
+import com.facebook.presto.sql.tree.IfExpression;
+import com.facebook.presto.sql.tree.Literal;
 import com.facebook.presto.sql.tree.Node;
-import com.facebook.presto.sql.tree.QualifiedName;
+import com.facebook.presto.sql.tree.OrderBy;
 import com.facebook.presto.sql.tree.Query;
+import com.facebook.presto.sql.tree.SearchedCaseExpression;
+import com.facebook.presto.sql.tree.SimpleCaseExpression;
+import com.facebook.presto.sql.tree.SortItem;
 import com.facebook.presto.sql.tree.SubqueryExpression;
+import com.facebook.presto.sql.tree.Window;
+import com.google.common.base.Splitter;
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMultimap;
+import com.google.common.collect.Multimap;
 
 import java.util.ArrayList;
-import java.util.Arrays;
+import java.util.Collection;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
+import java.util.Optional;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
-import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.facebook.presto.verifier.framework.VerifierUtil.PARSING_OPTIONS;
+import static com.google.common.base.Preconditions.checkState;
 import static java.util.Objects.requireNonNull;
 
 public class FunctionCallRewriter
-        extends DefaultTreeRewriter<FunctionCallRewriter.Context>
 {
-    public static class FunctionCallSubstitute
+    private static final List<Class<?>> SUPPORTED_ORIGINAL_FUNCTIONS = ImmutableList.of(FunctionCall.class, CurrentTime.class);
+    private static final List<Class<?>> SUPPORTED_SUBSTITUTE_EXPRESSIONS = ImmutableList.of(FunctionCall.class, IfExpression.class, SimpleCaseExpression.class,
+            SearchedCaseExpression.class, Identifier.class, Literal.class, ArrayConstructor.class);
+    private static final String OMIT_IDENTIFIER = "_";
+
+    private final Multimap<String, FunctionCallSubstitute> functionCallSubstituteMap;
+    private final FunctionAndTypeManager functionAndTypeManager;
+
+    private RewriterContext context = new RewriterContext();
```

Comment:
nit: final

Reply:
Oh, I have code like below
```java
class FunctionCallRewriter
{
    private RewriterContext context = new RewriterContext();
    ....
    public Node rewrite(Node root)
    {
        context = new RewriterContext();
        return new Rewriter(functionCallSubstituteMap, functionAndTypeManager).process(root, context);
    }
}
```
Would like each call of `functionCallRewriter.rewrite()` to have a new RewriterContext instance. RewriteContext at the moment is just used for keeping track of the actual function texts that are substituted and what they are substituted with.

Reply:
No need to keep it as a field anymore after taking your suggestion from another comment.